src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,92 @@
+package duke.task;
+
+import duke.exception.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a deadline object.
+ * It has an end date and time.
+ */
+public class Deadline extends Task {
+    private final LocalDate deadlineDate;
+
+
+    private static String[] parseDeadline(String task) throws DukeException {
+        String [] taskSplit = task.split("/by", 2);
+        String taskName = taskSplit[0].trim();
+
+        if (taskName.isEmpty()) {
+            throw new DukeException("Please enter task name");
+        }
+
+        if (taskSplit.length != 2) {
+            throw new DukeException("Please enter valid deadline (make sure to start /by)");
+        }
+
+        String dueDate = taskSplit[1].trim();
+        if (dueDate.isEmpty()) {
+            throw new DukeException("Please enter valid deadline: Do not leave it empty");
+        }
+
+        return new String[] {taskSplit[0], dueDate};
+    }
+
+    /**
+     * Initialize Deadline object that models a deadline
+     * @param task the deadline task name with a "/by" separating the description and the deadline
+     * @throws DukeException If the description or deadline is empty or there is no "/by" or
+     * the datetime format is wrong (not YYYY-MM-DD)
+     */
+    public Deadline(String task) throws DukeException {
+        super(parseDeadline(task)[0]);
+        try {
+            this.deadlineDate = LocalDate.parse(parseDeadline(task)[1]);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("wrong date time format: Use YYYY-MM-DD\n");
+        }
+
+    }
+
+    private Deadline(String task, boolean isDone, LocalDate deadlineDate) {
+        super(task, isDone);
+        this.deadlineDate = deadlineDate;
+    }
+
+    /**

```

Comment:
Is it necessary to return a new object everytime a change is made to the original object?

Reply:
Thanks for your input!